nullaway/src/test/java/com/uber/nullaway/NullAwayCustomLibraryModelsTests.java

```diff
@@ -227,4 +227,34 @@ public void libraryModelsAndSelectiveSkippingViaCommandLineOptions2() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void libraryModelsAndOverridingFieldNullability() {
+    makeLibraryModelsTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.lib.unannotated.UnannotatedWithModels;",
+            "public class Test {",
+            "   UnannotatedWithModels uwm = new UnannotatedWithModels();",
+            "   Object nonnullField = new Object();",
+            "   void assignNullableFromLibraryModelField() {",
+            "      // BUG: Diagnostic contains: assigning @Nullable",
+            "      this.nonnullField = uwm.nullableFieldUnannotated1;",
+            "      // BUG: Diagnostic contains: assigning @Nullable",
+            "      this.nonnullField = uwm.nullableFieldUnannotated2;",
+            "   }",
+            "   void flowTest() {",
```

Comment:
Can we also test that dereferencing a library field modeled as `@Nullable` causes NullAway to report an error?

Reply:
Sure, I thought it might be a good idea to add a fresh method in the same test for dereference:
0f73d506c657bea941a9a22147e3049af1ee966b